### PR TITLE
Turn setting Kokkos_CXX_STANDARD into an error

### PR DESCRIPTION
**Pull request by masterleinad**
Addressing https://github.com/kokkos/kokkos/pull/5283#discussion_r932847113.

### CODE REVIEW

File: cmake/kokkos_pick_cxx_std.cmake [hunk 1]
```diff
@@ -8,39 +8,12 @@ SET(KOKKOS_ENABLE_CXX14 OFF)
...
-  MESSAGE(STATUS "Setting default Kokkos CXX standard to ${CMAKE_CXX_STANDARD}")
+ELSE()
   SET(KOKKOS_CXX_STANDARD ${CMAKE_CXX_STANDARD})
 ENDIF()
-
-
-
-
-
+MESSAGE(STATUS "Setting default Kokkos CXX standard to ${KOKKOS_CXX_STANDARD}")
```
Comment (dalg24) [inline cmake/kokkos_pick_cxx_std.cmake:19]: The message is misplaced it should be in the `IF (NOT CMAKE_CXX_STANDARD)` branch

Comment (masterleinad) [reply cmake/kokkos_pick_cxx_std.cmake:19]: We were printing in both cases before and I don't see a problem with that.

Comment (masterleinad) [reply cmake/kokkos_pick_cxx_std.cmake:19]: If you insist, request changes.

**Review by dalg24 — APPROVED** [review]

**Review by nliber — APPROVED** [review]

File: cmake/kokkos_pick_cxx_std.cmake [hunk 2]
```diff
@@ -8,39 +8,12 @@ SET(KOKKOS_ENABLE_CXX14 OFF)
...
-  ELSEIF(${KOKKOS_CXX_STANDARD} STREQUAL "c++2a")
-    MESSAGE(WARNING "Deprecated Kokkos C++ standard set as 'c++2a'. Use '2A' instead.")
-    SET(KOKKOS_CXX_STANDARD "2A")
-  ENDIF()
+  MESSAGE(FATAL_ERROR "Setting the variable Kokkos_CXX_STANDARD in configuration is deprecated - set CMAKE_CXX_STANDARD directly instead")
 ENDIF()
 
-IF (NOT KOKKOS_CXX_STANDARD AND NOT CMAKE_CXX_STANDARD)
-  MESSAGE(STATUS "Setting default Kokkos CXX standard to 14")
+IF (NOT CMAKE_CXX_STANDARD)
```
Comment (ajpowelsnl) [inline cmake/kokkos_pick_cxx_std.cmake:14]: Would it be more intuitive to flip this logic, instead saying:

`IF (CMAKE_CXX_STANDARD)
   SET(KOKKOS_CXX_STANDARD ${CMAKE_CXX_STANDARD})
ELSE()
  SET(KOKKOS_CXX_STANDARD "14")
`

File: cmake/kokkos_pick_cxx_std.cmake (hunk 1 above)
Comment (ajpowelsnl) [reply cmake/kokkos_pick_cxx_std.cmake:19]: I viewed it in the same way as @masterleinad :  you have the message whatever the `CXX_STANDARD`

File: cmake/kokkos_pick_cxx_std.cmake (hunk 2 above)
Comment (masterleinad) [reply cmake/kokkos_pick_cxx_std.cmake:14]: I don't see much of a difference honestly.

**Pull request merged**